src/lightning/pytorch/utilities/combined_loader.py

```diff
@@ -289,21 +323,15 @@ def __next__(self) -> Any:
 
     def __iter__(self) -> Self:
         cls = _SUPPORTED_MODES[self._mode]["iterator"]
-        iterator = cls(self.flattened)
+        iterator = cls(self.flattened, self._limits)
         iter(iterator)
         self._iterator = iterator
         return self
 
     def __len__(self) -> int:
         """Compute the number of batches."""
-        lengths = []
-        for dl in self.flattened:
-            length = sized_len(dl)
-            if length is None:
-                raise NotImplementedError(f"`{type(dl).__name__}` does not define `__len__`")
-            lengths.append(length)
-        fn = _SUPPORTED_MODES[self._mode]["fn"]
-        return fn(lengths)
+        assert self._iterator is not None
```

Comment:
Since this is now a breaking change, can you mention it in the changelog and do

```suggestion
        if self._iterator is None:
            raise RuntimeError("Please call `iter(combined_loader)` first")
```